### ISSUE TITLE
syncStatusBar doesn't handle non-hex css

### ISSUE THREAD

**Issue by arnotixe** [aparajita/capacitor-dark-mode#7]
Problem:
* I have a gradient specified for **ion-content** `--background`, but capacitor Statusbar can't be set to that, it only takes hex colors.
* I can set the Statusbar to a compatible myself when I set the app theme, but then it won't sync on dark mode switch. Pink statusbar + dark mode app = no love. Really want dark-switch to sync the statusbar too, even in my

maybe an override css key and something like this in `src/base.ts` line 163ish?:
```ts
 if (content) {

        // Suggesting some kind of override read from ion-content or somewhere else
        const overrideBodyBackgroundColor = getComputedStyle(content)
        .getPropertyValue('--statusbarBackground')
        .trim()

        // or fallback to regular --background
        const bodyBackgroundColor = overrideBodyBackgroundColor || getComputedStyle(content)
          .getPropertyValue('--background')
          .trim()

          if (bodyBackgroundColor) {
          if (this.syncStatusBar !== 'textOnly') {
            await StatusBar.setBackgroundColor({
              color: normalizeHexColor(bodyBackgroundColor)
            })
          }

          setStatusBarStyle = true
        }
      }
```

Would be cool to override the statusbar style maybe (dark/light), but that should maybe be a different issue? It kind of would fit nicely into the same piece of code though. My app theme auto-calculates the text color (black or white, depending on background color lightness), so I have a css key for that. I use something like this:
```ts
StatusBar.setStyle({
      style:
        newcss['--textColor'] === '#000000'
          ? Style.Dark
          : Style.Light,
    });
```

**Comment by aparajita** [comment]
Feel free to submit a PR.

**Comment by arnotixe** [comment]
ok I setup #8 but still need to figure out how to actually test it in my local environment. Any hints on that?  :innocent: :pray:
(tried npm build on my local fork and syncing in the resulting dist/ folder into my app, but seems it doesn't pick up :thinking: )

**Comment by aparajita** [comment]
OK, now I understand what you're trying to achieve. I will implement it.

**Comment by arnotixe** [comment]
nice, I found on https://devdactic.com/build-capacitor-plugin how to install my local fork, just replace in package.json in the app:
```
"@aparajita/capacitor-dark-mode": "file:///path/to/capacitor-dark-mode",
```
and then
`npm i`
I can confirm the PR #8 actually works on android - also added some logic to automatically select white/black text on android status bar, based on the applied background color.

Of course you might want different wording or code style :grin: 

This is how I init it in one app, and the statusbar is kept in sync: (yes, I'm actually using `--ion-color-secondary` as background color there :man_shrugging: )
```
DarkMode.init({
      syncStatusBar: true,
      statusBarHexColorCssVariable: '--ion-color-secondary',
      getter: this.getAppearancePref,
    });
```

**Comment by aparajita** [comment]
I am working on a more comprehensive solution, thanks for the suggestion.

**Comment by aparajita** [comment]
Everything is implemented in Version 3.2.
